cpp/src/arrow/type.h

```diff
@@ -915,6 +926,16 @@ class ARROW_EXPORT Decimal128Type : public DecimalType {
 };
 
 /// \brief Concrete type class for 256-bit decimal data
+///
+/// Arrow decimals are fixed-point decimal numbers encoded as a scaled
+/// integer.  The precision is the number of significant digits that the
+/// decimal type can represent; the scale is the number of digits after
+/// the decimal point (note the scale can be negative).
+///
+/// For most use cases, the maximum precision offered by Decimal128Type
+/// is sufficient, and it will result in a more compact and more efficient
+/// encoding.  Decimal256Type is useful if you need a precision higher
+/// than 38 significant digits.
```

Comment:
Would it be possible to mention the 38 sig fig limit in 128-bit decimal documentation above (and if I'm calculating this correctly that the precision limit for 256-bit decimals is 76)?